plutus-core/plutus-core/src/PlutusCore/DeBruijn/Internal.hs

```diff
@@ -164,15 +164,14 @@ instance Wrapped TyDeBruijn
 
 instance (HasPrettyConfigName config) => PrettyBy config NamedDeBruijn where
   prettyBy config (NamedDeBruijn txt (Index ix))
-    -- See Note [Pretty-printing names with uniques]
-    | showsUnique = pretty . toPrintedName $ txt <> "_i" <> render (pretty ix)
+    | showsUnique = pretty $ toPrintedName txt <> "#" <> render (pretty ix)
```

Comment:
Oh, I really do believe we shouldn't use `#` for both uniques and De Bruijn indices. With the former `x#1` and `x#0` cannot be the same variable, but with the latter they can. We can have e.g. `#i` or something for De Bruijn indices.